sound/soc/sof/core.c

```diff
@@ -55,6 +56,26 @@ static const struct sof_panic_msg panic_msg[] = {
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_CLIENT)
 static int snd_sof_register_clients(struct snd_sof_dev *sdev)
 {
+	int i;
+	int ret;
+
+	/* Register platform independent client devices */
+	if (IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST)) {
+		for (i = 0; i < CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_NUM; i++) {
```

Comment:
Maybe create a function snd_sof_register_flood_clients and make it empty if SOF_DEBUG_IPC_FLOOD_TEST is not enabled. Otherwise, snd_sof_register_clients will be very hard to read.

Reply:
@dbaluta , yes that's an option. I'm not sure how it will look like if we pile up more platform independent clients like this.
The error handling might be much simpler with a separate function for each selectable client.

Reply:
@dbaluta, you are right, it is going to make it much cleaner, I'll update for v2.

Reply:
It's probably also worth a comment stating the future flow i.e. that capabilities will be read from FW manifest and devices enumerated as found in the manifest.

Reply:
I would add the comment when we have the code which does that